Assignment3/Q1.py

```diff
@@ -0,0 +1,128 @@
+"""The approach taken in this solution is based on a DFS-like search. The worst case running time
+   is exponential but because of the optimization used with isPrefix(string) method the average
+   running time should be a lot better.
+   The solution given below uses create_corresponding_graph(rows, columns) method to create a graph corresponding
+   to the grid - 2 dimensional array given. It uses a adjacency list representation of a graph
+   (Python dictionary) with an edge between every corresponding two adjacent entries in a given grid.
+   Then using both the given grid and a corresponding graph the DFS-like search is performed on a graph
+   (using words_search_recursive method) but it searches ALL possible paths from each vertex (corresponding
+   to a letter) similar to creating a big search tree with all possible words. Using isPrefix(string) method
+   it 'cuts off' early all the branches of the search tree which is being created that cannot form words."""
+
+from Dictionary import Dictionary
+
+
+def create_corresponding_graph(rows, columns):
+    """Method takes two arguments and returns a adjacency list representation of the corresponding grid with given
+       number of rows and columns.
+
+        Args:
+            rows: integer, number of rows in the grid
+            columns: integer, number of columns in the grid
+
+        Returns:
+            a dictionary, adjacency list representation of the graph corresponding to the character grid given in
+            the problem. A vertex in a graph is an integer tuple which indicates the coordinates of an entry in
+            the grid.
+    """
```

Comment:
That's a really good point, it would make it a lot clearer! I have changed my implementation now and I'm not creating the graph at all but I'll definately add an example of a data-structure that I'm creating in my future solutions!